src/thin_client.rs

```diff
@@ -15,9 +15,10 @@ use std::net::{SocketAddr, UdpSocket};
 use transaction::Transaction;
 
 pub struct ThinClient {
-    pub addr: SocketAddr,
-    pub requests_socket: UdpSocket,
-    pub events_socket: UdpSocket,
+    addr: SocketAddr,
+    requests_socket: UdpSocket,
+    events_socket: UdpSocket,
+    events_addr: SocketAddr,
```

Comment:
Put that before `events_socket` so that it's consistent with the requests addr/socket pair?